### PR TITLE
Add providers vi_VN ( Viet Nam )

### LINKED ISSUE #2075: Add Provider for vi_VN (Vietnamese)
* Faker version: Any
* OS: Any

Currently, the Faker library does not provide support for Vietnamese locale (vi_VN). To improve the localization and make it more useful for Vietnamese users, we should add a provider for Vietnamese names and other locale-specific data.

### Steps to reproduce

```
from faker import Faker
fake = Faker("vi_VN")
```

### Expected behavior

The provider should generate realistic Vietnamese names and other locale-specific data, such as:

Vietnamese male and female first names
Vietnamese last names
Addresses formatted according to Vietnamese standards

### Actual behavior

Currently, the vi_VN locale does not exist, and attempting to use it results in errors or fallback data.

### PR DESCRIPTION
**Pull request by ntd1683**
### What does this change

Adds Vietnamese language support (vi_VN) with providers for address, color, company, currency, date_time, job, lorem, person, and phone.

### What was wrong

The project lacked support for the Vietnamese locale, limiting its ability to generate localized data.

### How this fixes it

Introduces Vietnamese-specific data for address, color, company, currency, date_time, job, lorem, person, and phone.

Fixes #2075

### CODE REVIEW

**Comment by ntd1683** [comment]
@joke2k I have added the Vietnamese language, I checked it but there is an error due to your old code. I hope you can look and approve, I see you don't have Vietnamese yet.

**Comment by stefan6419846** [comment]
The typing failures have already been discussed in another PR. Apart from this, could you please add some source indicators for the new data?

**Comment by ntd1683** [comment]
> Các lỗi đánh máy đã được thảo luận trong một PR khác. Ngoài ra, bạn có thể vui lòng thêm một số chỉ số nguồn cho dữ liệu mới không?

ok I will add the source right away

File: faker/providers/person/vi_VN/__init__.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+from .. import Provider as PersonProvider
+
+
+class Provider(PersonProvider):
+    formats_female = (
```
Comment (fcurella) [inline faker/providers/person/vi_VN/__init__.py:5]: We need data source for names. Please refer to our [coding style](https://github.com/joke2k/faker/blob/master/docs/coding_style.rst#data-sets)

**Review by fcurella — CHANGES_REQUESTED** [review]

**Comment by ntd1683** [comment]
@fcurella  Hello, do I need to edit anything else to contribute to your project? Because I see that in my country, many people use your products

**Review by fcurella — APPROVED** [review]

**Pull request merged**